File: Data Structures & Algorithms/products-of-array-discluding-self/submission-17.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        product, zero_count = 1, 0

        for n in nums:
            if n:
                product *= n
            else:
                zero_count += 1
        if zero_count > 1: return [0] * len(nums)

        res = [0] * len(nums)

        for i, n in enumerate(nums): 
            if zero_count: res[i] = 0 if n else product
            else: res[i] = product // n
        return res
        

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-17.cs
public class Solution {
    public int FindMin(int[] nums) {
        int l = 0, r = nums.Length - 1;

        while(l < r) {
            int m = (l + r) / 2;
            if(nums[m] < nums[r]) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        return nums[l];
    }
}
